spec/lib/core/interactor/video_reader_spec.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'shared_examples/optional_failure_block'
+
+module Core
+  RSpec.describe VideoReader do
+    subject { described_class.new(id) }
+
+    let(:id) { 'the-video' }
+
+    describe '.call' do
+      before do
+        allow(Registry::Repository).to receive(:[]).with(:video) do
+          double(find: data)
+        end
+      end
+
+      context 'when the repository returns no data' do
+        let(:data) { nil }
+
+        it_has_behavior 'optional failure block'
+      end
+
+      context 'when the repository returns data' do
+        let(:title) { 'The Video' }
+        let(:description) { 'The Video has a description' }
+        let(:body) { '<h1>The Video</h1><p>Lorem ipsum dolor sit amet</p>' }
+        let(:categories) { [] }
+
+        let(:data) do
+          { title: title, description: description, body: body, categories: categories }
+        end
+
+        it 'instantiates the video with the id and the attributes from the repository' do
+          expect(Video)
+            .to receive(:new).with(id, data).and_call_original
+
+          subject.call
+        end
+
+        context 'when the Video entity is valid' do
+          before do
+            expect_any_instance_of(Video).to receive(:valid?) { true }
```

Comment:
This should be `allow_` and also if we can avoid stubbing any instance that would always be preferable.